crates/backend/src/ast.rs

```diff
@@ -475,8 +475,10 @@ pub struct Enum {
 #[cfg_attr(feature = "extra-traits", derive(Debug, PartialEq, Eq))]
 #[derive(Clone)]
 pub struct Variant {
-    /// The name of this variant
-    pub name: Ident,
+    /// The name of this variant in Rust
+    pub rust_name: Ident,
+    /// The name of this variant in JS
+    pub js_name: String,
```

Comment:
Could we just keep storing the `Ident` and `unraw()` it when we need it?

Reply:
This is not how something like that should be solved though imo.

If you see this as such a problem, then wrap the `Ident` into a new type, e.g. `Name`, and make two methods: `rust_name()` and `js_name()`.

Reply:
Most places where the `unraw` is missing are where an `Ident` is taken directly from the syn AST and then converted to a string for our backend AST. A newtype wouldn't solve that.

Reply:
I'm not sure what you are trying to say.

I imagined that you can encapsulate the state when `rust_name` and `js_name` is the same and when its not. How is that related to where a conversion happens?

Reply:
Okay, so this place here is the only place in the AST whether the JS identifiers are always only derived from Rust identifiers. This is only the case, because enum variants don't support `js_name` (yet). For all other cases, we already store the rust name as an `Ident` and the JS name as a `String`.

When I said that there are still many places left that don't `unraw`, I meant that there are still code paths deriving the JS name that don't account for this. Example [1](https://github.com/rustwasm/wasm-bindgen/blob/c60e8071a174489ac9b63ef40fa234f334034f85/crates/macro-support/src/parser.rs#L355) [2](https://github.com/rustwasm/wasm-bindgen/blob/c60e8071a174489ac9b63ef40fa234f334034f85/crates/macro-support/src/parser.rs#L580).

Reply:
> Okay, so this place here is the only place in the AST whether the JS identifiers are always only derived from Rust identifiers. This is only the case, because enum variants don't support `js_name` (yet). For all other cases, we already store the rust name as an `Ident` and the JS name as a `String`.

I see, a pointless optimization then.
Thank you for explaining.